### ISSUE TITLE
Correctly handle concurrency on same run id

### ISSUE THREAD

**Issue by lukaszachy** [psss/tmt#1269]
Not sure how much is this supported but at least for the second command to be just login it is something tempting to do (and working if timed correctly)

Reproducer
```
tmt -ddd -vvv run -i bug provision -h virtual -i fedora login -c 'echo FIRST; sleep 1000' |& tee /tmp/first &
sleep 20
tmt -ddd -vvv run -i bug login -c 'echo SECOND' |& tee /tmp/second
```

On my machine, "second" tmt command (the one which is supposed to just login) generates ssh private key and continues to login but by doing so it locks out first command which waits asking for password until timeout happens.

Probably the only thing tmt should allow is 'standalone' login mode. All other attempts to use run from active workdir should be caught and denied unless user to removes the lock.

**Comment by idorax** [comment]
> Probably the only thing tmt should allow is 'standalone' login mode. All other attempts to use run from active workdir should be caught and denied unless user to removes the lock.

@lukaszachy, if we create a lock file for `login`,  the lock file would be saved as `WORKDIR_ROOT/<GuestID>.lock` on host side （e.g. `/var/tmp/tmt/local.lock` if  local host is used as `guest` for test execution), is it okay?

**Comment by lukaszachy** [comment]
I'm not sure it is worth doing right now before multihost which will require a lot of changes in the flow of tmt run.

There are corner cases to solve for better UX
- If there is just 'login' then skip running all steps but login, with question about whether to do the `push` or not (I think both use cases are valid and user should be able to choose)
- Should wait until ready to login, we can't tell user "it's too early to login, try again later"

**Comment by idorax** [comment]
> I'm not sure it is worth doing right now before multihost which will require a lot of changes in the flow of tmt run.
OK, let's not try to fix it for the time being.

**Comment by psss** [comment]
Yeah, when the `provision` step is not finished another `tmt run` command should probably just wait. When an existing `tmt` process is in progress, it really seems that `login` would be the only action which makes sense, not to break anything which is currently being worked on in the first process. Adding to the backlog.
